app/models/message.rb

```diff
@@ -50,14 +52,22 @@ def read!
     update! read: true
   end
 
-  def toggle_approved!
-    toggle! :approved
+  def toggle_approved!(user)
```

Comment:
Why not having separate methods for this? Could we delegate this to `approve!` and `disapprove!`?